### PR TITLE
Add a function to run arbitrary sql with arguments.

### PR DESCRIPTION
**Pull request by montanalow**
This helps to ensure args are properly encoded, rather than relying on mechanisms like format!() that may be less correct or secure.

Follow up to #682 with proper git branching.

### CODE REVIEW

**Comment by workingjubilee** [comment]
```
    left: `Array([Object({"Plan": Object({"Async Capable": Bool(false), "Node Type": String("Result"), "Parallel Aware": Bool(false), "Plan Rows": Number(1), "Plan Width": Number(1), "Startup Cost": Number(0.0), "Total Cost": Number(0.01)})})])`,
    right: `Array([Object({"Plan": Object({"Node Type": String("Result"), "Parallel Aware": Bool(false), "Plan Rows": Number(1), "Plan Width": Number(1), "Startup Cost": Number(0.0), "Total Cost": Number(0.01)})})])`
```
The fields of Explain can vary based on PG version. You may want to match on the fields you know will be present in all versions. I will also accept `#[cfg(feature = "pg__")]`

**Comment by workingjubilee** [comment]
Alright. This looks good. But, while I do not mean to keep this long, this isn't a bugfix or a critical break, and I need time to think on whether there needs to be any differences w/r/t this API before I can merge it, seeing as how you will be immediately depending on it (probably). So it looks like this will be 0.5.1

**Comment by workingjubilee** [comment]
We can merge this but @eeeebbbbrrrr mentioned not feeling great about the general pattern our interfaces have taken here, so I want to doublecheck with him before doing so.

File: pgx-tests/src/tests/spi_tests.rs [hunk 1]
```diff
@@ -124,6 +124,47 @@ mod tests {
...
+    #[pg_test]
+    fn test_spi_run_with_args() {
+        let i = 1 as i32;
+        let j = 2 as i64;
+
+        Spi::run_with_args(
+            "SELECT $1 + $2 = 3",
+            Some(vec![
+                (PgBuiltInOids::INT4OID.oid(), i.into_datum()),
+                (PgBuiltInOids::INT8OID.oid(), j.into_datum()),
```
Comment (workingjubilee) [inline pgx-tests/src/tests/spi_tests.rs:141]:
```suggestion
                (PgBuiltInOids::INT4OID.oid(), Some(i.into()),
                (PgBuiltInOids::INT8OID.oid(), Some(j.into())),
```

File: pgx-tests/src/tests/spi_tests.rs [hunk 2]
```diff
@@ -124,6 +124,47 @@ mod tests {
...
+    #[pg_test]
+    fn test_spi_explain_with_args() {
+        let i = 1 as i32;
+        let j = 2 as i64;
+
+        let result = Spi::explain_with_args(
+            "SELECT $1 + $2 = 3",
+            Some(vec![
+                (PgBuiltInOids::INT4OID.oid(), i.into_datum()),
+                (PgBuiltInOids::INT8OID.oid(), j.into_datum()),
```
Comment (workingjubilee) [inline pgx-tests/src/tests/spi_tests.rs:161]:
```suggestion
                (PgBuiltInOids::INT4OID.oid(), Some(i.into()),
                (PgBuiltInOids::INT8OID.oid(), Some(j.into())),
```

**Review by workingjubilee — COMMENTED** [review]
Whoops, sorry. Imports shifted around.

**Review by eeeebbbbrrrr — APPROVED** [review]
LGTM!

Thanks!

File: pgx-tests/src/tests/spi_tests.rs [hunk 3]
```diff
@@ -124,6 +124,47 @@ mod tests {
...
+
+    #[pg_test]
+    fn test_spi_explain_with_args() {
+        let i = 1 as i32;
+        let j = 2 as i64;
+
+        let result = Spi::explain_with_args(
+            "SELECT $1 + $2 = 3",
+            Some(vec![
+                (PgBuiltInOids::INT4OID.oid(), Some(i.into()),
```
Comment (workingjubilee) [inline pgx-tests/src/tests/spi_tests.rs:160]:
```suggestion
                (PgBuiltInOids::INT4OID.oid(), Some(i.into())),
```

File: pgx-tests/src/tests/spi_tests.rs [hunk 4]
```diff
@@ -124,6 +124,47 @@ mod tests {
...
+
+    #[pg_test]
+    fn test_spi_run_with_args() {
+        let i = 1 as i32;
+        let j = 2 as i64;
+
+        Spi::run_with_args(
+            "SELECT $1 + $2 = 3",
+            Some(vec![
+                (PgBuiltInOids::INT4OID.oid(), Some(i.into()),
```
Comment (workingjubilee) [inline pgx-tests/src/tests/spi_tests.rs:140]:
```suggestion
                (PgBuiltInOids::INT4OID.oid(), Some(i.into())),
```

**Review by workingjubilee — COMMENTED** [review]
Amazing, I can't believe I missed that.

**Comment by workingjubilee** [comment]
I will rustfmt this \_\_\_\_\_\_\_\_\_\_\_\_\_ myself.

**Pull request merged**